fluentforms/core/src/test/java/com/_4point/aem/fluentforms/testing/assembler/MockAssemblerServiceTest.java

```diff
@@ -1,14 +1,38 @@
 package com._4point.aem.fluentforms.testing.assembler;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Map;
+
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com._4point.aem.fluentforms.api.Document;
+import com._4point.aem.fluentforms.api.assembler.AssemblerResult;
+import com._4point.aem.fluentforms.testing.MockDocumentFactory;
+import com._4point.aem.fluentforms.testing.assembler.MockTraditionalAssemblerService.GenerateAssemblerResultArgs;
+
 
 public class MockAssemblerServiceTest {
 	
+	@SuppressWarnings("unchecked")
 	@Test
 	void testInvokeMockDocument() throws Exception {
+		String expectedResultString = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?><DDX xmlns=\"http://ns.adobe.com/DDX/1.0/\">\r\n" + 
+				"<PDF result=\"concatenatedPDF.pdf\"><PDF source=\"File0.pdf\"/>\r\n" + 
+				"<PDF source=\"File1.pdf\"/></PDF></DDX>";
+		MockDocumentFactory mockDocumentFactory = new MockDocumentFactory();
+		Document ddx = mockDocumentFactory.create(expectedResultString.getBytes());
+		Map<String,Object>sourceDocuments =  Mockito.mock(Map.class);
 		
+		MockTraditionalAssemblerService mockTraditionalAssemblerService = new MockTraditionalAssemblerService() ;
+	    MockAssemblerService underTest = MockAssemblerService.createAssemblerResultMock(mockTraditionalAssemblerService.getDUMMY_ASSEMBLER_RESULT());
```

Comment:
This test could be improved.  If the user doesn't set the AssemblerResult ahead of time, the MockTraditionalAssemblerService will return the DUMMY_ASSEMBLER_RESULT.  By using the DUMMY_ASSEMBLER_RESULT for this test, you are not really able to tell whether the routine is returning DUMMY_ASSEMBLER_RESULT because that's what you passed in or whether it is returning DUMMY_ASSEMBLER_RESULT because it thinks you didn't pass anything in.  You should pass in something else that *doesn't* match DUMMY_ASSEMBLER_RESULT and then test that the result matches what you passed in (and therefore doesn't *isn't* DUMMY_ASSEMBLER_RESULT).

This, plus the additional test I mentioned above to test for DUMMY_ASSEMBLER_RESULT when nothing has been specified should cover both branches of the terniary statement in `MockTraditionalAssemblerService`.

Reply:
Added test which take assembler result that does not match DUMMY_ASSEMBLER_RESULT.